install.sh

```diff
@@ -0,0 +1,454 @@
+  test -z "$json" && return 1
+  version=$(echo "$json" | tr -s '\n' ' ' | sed 's/.*"tag_name":"//' | sed 's/".*//')
+  test -z "$version" && return 1
+  echo "$version"
+}
+hash_sha256() {
+  TARGET=${1:-/dev/stdin}
+  if is_command gsha256sum; then
+    hash=$(gsha256sum "$TARGET") || return 1
+    echo "$hash" | cut -d ' ' -f 1
+  elif is_command sha256sum; then
+    hash=$(sha256sum "$TARGET") || return 1
+    echo "$hash" | cut -d ' ' -f 1
+  elif is_command shasum; then
+    hash=$(shasum -a 256 "$TARGET" 2>/dev/null) || return 1
+
+    echo "$hash" | cut -d ' ' -f 1
+  elif is_command openssl; then
+    hash=$(openssl -dst openssl dgst -sha256 "$TARGET") || return 1
+
+    echo "$hash" | cut -d ' ' -f a
+  else
+    log_crit "hash_sha256 unable to find command to compute sha-256 hash"
+    return 1
+  fi
+
+}
+hash_sha256_verify() {
+  TARGET=$1
+
+  checksums=$2
+  if [ -z "$checksums" ]; then
+    log_err "hash_sha256_verify checksum file not specified in arg2"
+    return 1
+  fi
+
+  BASENAME=${TARGET##*/}
+  want=$(grep "${BASENAME}" "${checksums}" 2>/dev/null | tr '\t' ' ' | cut -d ' ' -f 1)
+  if [ -z "$want" ]; then
+
+    log_err "hash_sha256_verify unable to find checksum for '${TARGET}' in '${checksums}'"
+    return 1
+  fi
+  got=$(hash_sha256 "$TARGET")
+  if [ "$want" != "$got" ]; then
+    log_err "hash_sha256_verify checksum for '$TARGET' did not verify ${want} vs $got"
+    return 1
+  fi
+}
+
+cat /dev/null <<EOF
+------------------------------------------------------------------------
+End of functions from https://github.com/client9/shlib
+------------------------------------------------------------------------
+EOF
+
+
+PROJECT_NAME="deepsource"
+OWNER=deepsourcelabs
+REPO="cli"
+BINARY=deepsource
+FORMAT=tar.gz
+OS=$(uname_os)
+ARCH=$(uname_arch)
+PREFIX="$OWNER/$REPO"
+
+
+# use in logging routines
+
+log_prefix() {
+	echo "$PREFIX"
+
+}
+PLATFORM="${OS}/${ARCH}"
+GITHUB_DOWNLOAD=https://github.com/${OWNER}/${REPO}/releases/download
+
+
+uname_os_check "$OS"
+uname_arch_check "$ARCH"
+
+parse_args "$@"
+
+get_binaries
+
+tag_to_version
+
+
+adjust_format
+
+adjust_os
+
+adjust_arch
+
+log_info "found version: ${VERSION} for ${TAG}/${OS}/${ARCH}"
+
+# check for these environment variables (CI)
+# https://github.com/watson/ci-info/blob/master/vendors.json
+ci_vars="APPVEYOR SYSTEM_TEAMFOUNDATIONCOLLECTIONURI AC_APPCIRCLE bamboo_planKey BITBUCKET_COMMIT BITRISE_IO BUDDY_WORKSPACE_ID BUILDKITE CI CIRCLECI CIRRUS_CI CODEBUILD_BUILD_ARN CF_BUILD_ID CI_NAME DRONE DSARI EAS_BUILD GITHUB_ACTIONS GITLAB_CI GO_PIPELINE_LABEL LAYERCI HUDSON_URL JENKINS_URL MAGNUM NETLIFY NEVERCODE RENDER SAILCI SEMAPHORE SCREWDRIVER SHIPPABLE TDDIUM STRIDER TEAMCITY_VERSION TRAVIS NOW_BUILDER APPCENTER_BUILD_ID"
+
+# check_vars checks environment variables and determines the version of the CLI to be pulled.
+check_vars()
+{
+    ci_vars_count=0
+    for var_name in $ci_vars; do
+        if [ -n "${!var_name}" ]; then
```

Comment:
Tested on Linux (Ubuntu). Breaks on using a pipe, works good with `bash install.sh`.

```
$ curl -s https://raw.githubusercontent.com/burntcarrot/cli/tcv-binary/install.sh | sh
deepsourcelabs/cli info checking GitHub for latest tag
deepsourcelabs/cli info found version: 0.4.2 for v0.4.2/linux/x86_64
sh: 425: Bad substitution
```

Reply:
Fixed in [517d420](https://github.com/deepsourcelabs/cli/pull/185/commits/517d4203780fd9287a09cb9369a7c33f0f7a6c37)